### DISCUSSION TITLE
Require whitelisting for Access Level 4 on Identity

### DISCUSSION THREAD

**Discussion by maebeam** [deso-protocol/dips#120] [Medium CIPs]
Today any third party app can ask users for Access Level 4 (full singing access) on BitClout Identity. Most users are not aware that this level of access is much more dangerous than Access Level 3 and a malicious app could steal all their funds. I propose a community-maintained whitelist of nodes that are allowed to request full signing access. Third party apps that want to be able to request full singing access will need to submit a pull request to the Identity repo and demonstrate their trustworthiness. Some rough guidelines should be established for how to demonstrate trustworthiness and community buy in.

Inspired by #117 from @trevormil

**Comment by dharmeshs** [comment]
From the cheap seats (I haven't contributed any code yet): This is a *really* good idea.

**Reply by trevormil** [reply]
Good idea in concept, but my issue lies with the centralization of this. Who gets to choose who is trusted? Community vote? How do they demonstrate trustworthiness? What is stopping someone from switching to the dark side when they see a user with $10 million worth of $CLOUT log into their app. Money changes people. You can trust them at the beginning, but anyone can gain trust just to get an approval.

And they have no consequences. With mainstream crypto if Coinbase for example acts untruthfully, they get multimillion dollar lawsuits. What is stopping some random person who started a new BitClout project from hacking funds? It isn't going to end well. I'm sure trust will develop over time with certain projects, but there needs to be some changes with BitClout Identity itself, not who gets access to it.

**Reply by trevormil** [reply]
And since "BitClout is not a company, just code" according to their docs, I'd assume there are no legal consequences here either.

**Reply by ItsAditya-xyz** [reply]
agreed with what @trevormil said! Not only this makes bitclout more centralised but also it offers some risk since false trust can be made and its consequences could be terrible.

**Comment by trevormil** [comment]
Good idea in concept, but I don't think this is the right way to go about it. This is just making BitClout more centralized which I also talked about in https://github.com/bitclout/cips/discussions/117. This needs to be really thought out and carefully implemented if this is the path that is chosen to be implemented. I don't have a specific technical solution right now. I'll think about it and get back to you. 

But one thing for certain is there needs to be more education to the average user on this. BitClout attracts average people because it is a social media platform, and most people aren't crypto savvy. I'd recommend making this more known to the average users.

**Reply by trevormil** [reply]
I have somewhat of an idea. Stake $CLOUT / creator coin for access level 4. If you act distrustfully, you lose the money. Proof of stake pretty much. Lost money can go to a developer pool that funds other projects.

**Reply by trevormil** [reply]
There needs to be some sort of consequence for access level 4 because right now, there is absolutely nothing stopping anyone from funneling funds. Community approval or a whitelist is not much of a consequence. And once it is time to vote on a user after they act untruthfully, they have already acted untruthfully and damage has been done.

**Reply by Dopelessperza** [reply]
I like the idea of proof of stake. Depending on the Mount that needs to be staked.

It's interesting because as a person who has recently spun up a node with minimal resources, the idea of having to stake a large amount of $clout is slightly terrifying.

But money talks that's for sure. It's an incentive to overcome obstacles that finances have for many.

The barrier of entry can keep out people who don't have skin in the game, people who only wish to take advantage. But it can also keep people out who don't have deep pockets yet have exceptional ideas.

**Comment by FreeTrade** [comment]
This is another centralizing move - identity.bitclout.com is centralized enough already - the direction should be towards decentralization.

I suggest more work on derived keys with different authorizations, and the ability of users to set some limit on level 4 transaction size/frequency. It's harder, I know, but BitClout raised a lot of fund on the promise of a decentralized system.

**Reply by trevormil** [reply]
Agreed. Every single decision they have been making for the past couple months, they have failed to keep in mind their core motto of who they are. To the core team, please always keep in mind that you are a decentralized protocol first and foremost, or at least that is what you claim to be (haven't seen any evidence of it though). It seems like you just don't care anymore about decentralization anymore. This is a mistake and I guarantee this centralization will cause BitClout a lot of damage further down the road

**Reply by arhebbar** [reply]
I agree with this suggestion and had posted about this earlier. The authorization should be up to certain levels that the user explicitly authorizes ONLY. That also addresses the point about educating the users on what they are authorizing the app with doing. 

_I authorize xyz.com node to go ahead with transactions up to a sum total of $100 without explicitly checking with me._

**Comment by iPaulPro** [comment]
While I understand the desire and value of this proposal, I'd prefer a visual indicator/badge for "verified" nodes; not restricting approvals only to those on the whitelist.

In my opinion, the only nodes that should be trusted with level 4 are ones that are fully open source, including the CD/release pipeline. I'd support a badge in Identity for nodes that satisfy this criteria, but not even bitclout.com currently does.

**Reply by Lux1n** [reply]
I'd like to see a trust system like this developed for nodes, maybe a visual badge AND a link to their code on Github (sorry I don't know the best term; repos?).

**Comment by marnimelrose** [comment]
Yes, I brought up the idea of "Bitclout powered" in #94. However, since we are an existing app that is not open source and we were looking at integrating then companies like ours might be out of the game if we are forced to go fully open source. I think this would hinder the potential growth.

**Reply by iPaulPro** [reply]
We're you responding to my comment? I don't believe this proposal mentions open source, and there was no talk of it being mandatory.

**Reply by marnimelrose** [reply]
Hi Paul, above didn't you say that the only nodes that should be trusted are the open source nodes? Did I misunderstand what you were saying?

**Reply by iPaulPro** [reply]
> Hi Paul, above didn't you say that the only nodes that should be trusted are the open source nodes? Did I misunderstand what you were saying?

Sorry, it wasn't clear because you responded to the main thread and didn't quote me. I explicitly stated that I believe the only thing that should happen here is a visual "badge" , and not restricting access to only those in this whitelist. However, I believe those badges should only be awarded to fully open source projects as they are the only ones that can truly be trusted.

**Reply by arhebbar** [reply]
As someone looking at building a node, I can say that any favoritism to open source platforms would not help this scale. What might be fair is to have a category for open-source nodes and another for those that are not. Users can then decide based on who is backing a closed project and what level of access is needed whether to give the said access or not. A visual badge is fine - maybe one for open source, one for a project that has gone thru a community audit or something like that. 

Users can and should be encouraged to hold their larger investments including their own coin in a secure wallet / id over time and move to the main id only for trading. While this may not work for the main use-case on BitClout.com or BitCloutPulse, most user-driven use-cases will not need major investments or a lot of money in their wallet and can give Level 4 access. Think of a Medium built on BitClout like CloutPub that @Thorsten is building.

**Reply by iPaulPro** [reply]
> I can say that any favoritism to open source platforms would not help this scale.
> A visual badge is fine - maybe one for open source, one for a project that has gone thru a community audit or something like that. 

Only thing I suggested is a visual badge. Audits for closed source software are useless. Please provide credentials and back up your claims that favoring open source (as all blockchain/crypto do) will not help this scale.

> Users can and should be encouraged to hold their larger investments including their own coin in a secure wallet / id over time

The platform is confusing enough and we're trying to attract the masses. I believe we should be working out derived keys that afford specific permissions, much like logging in with Facebook.

**Reply by tijno** [reply]
>  I believe we should be working out derived keys that afford specific permissions

YES! #115 and my comment here https://github.com/bitclout/core/pull/69#issuecomment-891136912

**Reply by arhebbar** [reply]
> > I can say that any favoritism to open source platforms would not help this scale.
> > A visual badge is fine - maybe one for open source, one for a project that has gone thru a community audit or something like that.
> 
> Only thing I suggested is a visual badge. Audits for closed source software are useless. Please provide credentials and back up your claims that favoring open source (as all blockchain/crypto do) will not help this scale.

Fair point on the audit of closed source software. Maybe, an open directory of all projects based on the transactions they are driving and in a way contributing to the network growth. 

Credentials: I am @Randhir on BitClout - 140+ Days, Top 125 in terms of engagement throughout and working on a closed source project called CreaTiers where the vision is to be the Upwork for BitClout and over time hopefully the node where everyone will come to discover and engage top-tier Creators. 

At least until we get some traction, we will most likely remain closed source. I personally am fairly new to the open source ecosystem and our decision on going open source vs. closed source is still not fully sorted.
